File: Semester-04/Personal/DNS.py
import webbrowser

def FindIP(Input):
 Link = {
  "yahoo.com"    : "74.6.231.20"     ,
  "youtube.com"  : "142.251.35.174"  ,
  "google.com"   : "142.251.40.174"  ,
  "facebook.com" : "57.144.180.1"    ,
  "instagram.com": "157.240.19.174"  ,
  "x.com"        : "162.159.140.229"
      }
 if Input == "|?":
  print("to get IP : example.com")
  print("to stop: exit")
  return ""
 
 elif Input == "exit":
  print("program exited successfully.")
  return "exit"
 

  
def OpenLink(IP):
 if IP == "" or IP == "exit": return

 Ans = input("Open IP?(YES OR NO)").strip().lower()

 if Ans == "yes":
  URL = "http://" + IP
  webbrowser.open(URL)
 elif Ans == "no": return
 else:
  print("Wrong input ")
  OpenLink(IP)

In = ""
while In != "exit":
 In =  input("(|? for help)(Insert the link): ").strip().lower()
 FindIP(In)
 
#remove function
#Add function
#Change function

File: Semester-04/Personal/DNS.cpp
#include <iostream>
#include <map>
#include <cstdlib>
#include <algorithm>
using namespace std;

string FindIP(string Input)
{
  map <string , string> Link;
 
  Link["yahoo.com"]    = "74.6.231.20"    ;
  Link["youtube.com"]  = "142.251.35.174" ;
  Link["google.com"]   = "142.251.40.174" ;
  Link["facebook.com"] = "57.144.180.1"   ;
  Link["instagram.com"]= "157.240.19.174" ;
  Link["x.com"]        = "162.159.140.229";            

  if (Link.find(Input) != Link.end())
  {
    cout << "The IP Is : " << Link[Input] << endl;
    return Link[Input];
  }
  else if (Input == "|?") 
  { 
        cout << "to get IP : example.com" << endl
             << "to stop: exit" << endl;
        
        return "";
  }
  else if (Input == "exit") 
  {
    cout << "Program exited successfully." << endl;
    return "exit";
  }
  else
  {
    cout << "wrong input. TRY AGAIN" << endl;
  }
}

void open_link(string IP)
{
    if (IP.empty() or IP == "exit") return;
    string Ans;
    cout << "open IP?(YES or NO)" << endl; getline(cin, Ans);
    transform(Ans.begin(), Ans.end(), Ans.begin(), ::tolower);

    if (Ans == "yes")
    {
      string command = "start http://" + IP;
      system(command.c_str());
    }
    else if(Ans == "no")
    {
      return;
    }
    else
    {
      cout << "wrong input " << endl;
      open_link(IP);
    }

}

int main() 
{
 string Input;
 while(Input != "exit")
 {
  cout << "(|? for help)(Insert the link):"; getline(cin , Input);
  transform(Input.begin(), Input.end(), Input.begin(), ::tolower);
  string IpAddress = FindIP(Input);
  if (!IpAddress.empty())
   open_link(IpAddress);
 }
}
//add function
//remove function
//change fuction
